### ISSUE TITLE
Sidenav: HuBMAP Apps

### ISSUE THREAD

**Issue by LibbyUX** [hubmapconsortium/hra-ui#557]
# Implement HRA version of HuBMAP shared navigation component
- Design menu options to pitch tomorrow at CNS UX specs
- Redesign every app header to feature this button with shared nav functionality

# Resources:
- [Data Portal](https://portal.hubmapconsortium.org/): Use nav header button Resources and Apps icon to see menu
- HIVE UX-UI Collaboration Spreadsheet: [HuBMAP Tools Text & Image Assets](https://docs.google.com/spreadsheets/d/12obX2cWD6Sw3L0IWXMs47LT4YyBJEKbxNUCcNLqM98w/edit?pli=1&gid=0#gid=0)

<img width="917" alt="Screenshot 2024-07-16 at 2 16 18 PM" src="https://github.com/user-attachments/assets/e41e32c9-2fe1-4c4a-8776-a32563c68fdc">

**Comment by LibbyUX** [comment]
# Status update:
- Moving all navigation areas to side nav for better mobile responsiveness. This feature will need to seamlessly fold into our existing and new side navs. Thank you @axdanbol for your help with this design!

### TO DO
Design updates for nav bars current UIs:
- HRA Portal current implementation (mobile side nav in use currently)
- Dashboards (mobile side nav in use currently)
- Cell Distance Explorer
- FTU Explorer
  
Passing this to Katy today for review on FTU Explorer & Atlas Apps.

**Comment by LibbyUX** [comment]
### Design Review
Passed to @katyb tonight for feedback or approval:
- Prototype: [HuBMAP Shared Navigation (Side nav overlay)](https://www.figma.com/proto/VlNjSrctLEc0Wyu32FmnZq/Functional-Tissue-Unit-(FTU)-Explorer?page-id=5919%3A26286&node-id=5919-33586&viewport=742%2C-2884%2C0.29&t=YEO5HPNGEEkWyXGJ-1&scaling=contain&content-scaling=responsive&starting-point-node-id=5919%3A33586&show-proto-sidebar=1)

<img width="1280" alt="Screenshot 2024-07-18 at 10 31 06 PM" src="https://github.com/user-attachments/assets/a7b76d8f-f197-494c-ab33-e0158124404e">

<img width="918" alt="Screenshot 2024-07-18 at 10 32 21 PM" src="https://github.com/user-attachments/assets/6d2758b3-3acf-4a82-8281-8cec9bf1c13c">

**Comment by katyb** [comment]
Text for
![image](https://github.com/user-attachments/assets/53e81b99-9a56-4f75-92ae-33226108cd9d)

is incorrect

**Comment by katyb** [comment]
Where can I copy the text to make changes without retyping?

**Comment by LibbyUX** [comment]
@katyb and I edit the text together in the collaborative HIVE UX-UI [HuBMAP Tools for Shared Navigation spreadsheet](https://docs.google.com/spreadsheets/d/12obX2cWD6Sw3L0IWXMs47LT4YyBJEKbxNUCcNLqM98w/edit?gid=0#gid=0) at 1:15pm.

@katyb, is this component approved now? Can you please provide approval in writing here on GitHub?

### Shared Nav: HuBMAP Data Portal vs. HRA Portal styles
- Design System Repo Canvas: [HuBMAP Nav Overlay Menu](https://www.figma.com/design/gQEMLugLjweDvbsNNUVffD/Design-System-Repository?node-id=4806-18532&t=4qnlcRfeB9QUaTNZ-4)

![image](https://github.com/user-attachments/assets/2e7402e0-6736-43a4-abd5-11a061277f25)

**Comment by LibbyUX** [comment]
All edits have been made. Just waiting on approval. Thanks!

**Comment by LibbyUX** [comment]
![image](https://github.com/user-attachments/assets/17268db2-a535-4e1a-a99e-910e02eb0389)

**Comment by LibbyUX** [comment]
This was approved today! More details coming soon!

![image](https://github.com/user-attachments/assets/1f5c13a0-f751-4283-ae2d-50b222d390d2)

**Comment by LibbyUX** [comment]
# Side Nav: HuBMAP Apps
- Design System Components: [Side Nav: HuBMAP Apps](https://www.figma.com/design/BCEJn9KCIbBJ5MzqnojKQp/Design-System-Components?node-id=78-355)
<img width="2068" alt="Screenshot 2024-07-25 at 4 27 12 PM" src="https://github.com/user-attachments/assets/f290733b-7e96-4658-a85e-35c7f0e5b307">

## Building Blocks
- Unique menu style:
  - Background blur
  - Lowered opacity
- #521 
- #586 
- #573 
- Fonts
  - Title font: Title Large
  - Section font: Title Medium
- Overlay background: #000000 at 48%
- Min/max widths to keep it responsive
- Perfect matching alignment with the #573 40x40 variant from the menu overlay to the nav header access point

## Usage
- HuBMAP's HIVE UX/UI Cross Collaboration team has created a menu to connect HuBMAP applications cohesively. This menu is accessed by clicking the apps icon button in our new navigation headers.

### Variants
- Ready/default
- Hover (activate scrollbar when lower content needs room to grow)

## Behavior
- This organism is responsive to smaller resolutions and should grow as specified with its min/max widths
-  #521 - Appears on hover, when content needs space to grow/scroll
- The bottom part of the menu will always stretch to fill the entire screen height
### Figma Interaction Specifications
<img width="2019" alt="Screenshot 2024-07-25 at 4 45 06 PM" src="https://github.com/user-attachments/assets/1d76dc04-9415-4c1f-807e-219476f64ab3">

**Comment by LibbyUX** [comment]
@axdanbol - Just wanted to bring this issue to your attention today. Happy Friday!!!

I will create a HuBMAP-styled version of this today for implementation on humanatlas.io immediately. This HuBMAP-styled variant will be deprecated after we implementing the redesigned HRA portal sometime in Y3.

I will only be changing the font type to Inter and I will change the font/icon colors probably to a unified #212121. The rest of the component is unchanged!

I will link this HuBMAP variant here when I have it!

**Comment by LibbyUX** [comment]
Update: I will open a separate issue for the HuBMAP styles app menu for the current humanatlas.io. This component is ready for development in our design system. Thanks!

**Comment by bhushankhope** [comment]
@LibbyUX . I have created a sidenav demo [here](https://github.com/hubmapconsortium/hra-ui/pull/655#issuecomment-2297153691). Please review. Thanks

**Comment by LibbyUX** [comment]
Updated issue here: #766
